src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -797,7 +797,9 @@ public void Connect(EndPoint remoteEP)
                 throw new InvalidOperationException(SR.net_sockets_mustnotlisten);
             }
 
-            if (_isConnected)
+            // We're ignoring datagram sockets even if it's connected.
+            // Because, we can disconnect some protocols by connecting another endpoint. (for ex: UDP)
+            if (_isConnected && _socketType != SocketType.Dgram)
```

Comment:
There are a bunch of other such checks, as introduced in https://github.com/dotnet/corefx/pull/27173.  This is the only one that needs special-casing?  And is Dgram the only special-case, or should this actually be inverted to only do the check if it's SocketType.Stream?

Reply:
This is not the only one, I'll add this relax it in every path.

[man connect](https://man7.org/linux/man-pages/man2/connect.2.html) page says: 
```
       Some protocol sockets (e.g., datagram sockets in the UNIX and
       Internet domains) may use connect() multiple times to change
       their association.

       Some protocol sockets (e.g., TCP sockets as well as datagram
       sockets in the UNIX and Internet domains) may dissolve the
       association by connecting to an address with the sa_family member
       of sockaddr set to AF_UNSPEC; thereafter, the socket can be
       connected to another address.  (AF_UNSPEC is supported on Linux
       since kernel 2.2.)
```

According to this, we need to remove it for every possible type of socket. But, after a second thought, I think it should be better to keep it only for `Stream` type sockets for now.

Reply:
I think the datagram behavior is somewhat obscure and my guess it that https://github.com/dotnet/corefx/pull/27173 was not aware of it e.g. it did what ever make sense for TCP. 

I'm not sure if there is practical difference but checking for `Stream` may make sense. AFAIK there is no use case for `SocketType.Raw`  but if the operation does not make sense I would expect OS to return error e.g. this was preventive check for better exception and perhaps consistency.

Reply:
I've updated the PR description with WinSock, Linux, and OSX API docs.